src/main/java/backend/Logic.java

```diff
@@ -247,13 +209,204 @@ public Model getRepo(String repoId) {
+     *
+     * @param filterTexts Filter expressions to process.
+     */
+    private void openRepositoriesInFilters(List<FilterExpression> filterTexts) {
+        HashSet<String> reposToOpen = new HashSet<>();
+
+        filterTexts.forEach(filterText -> {
+            filterText.find(Qualifier::isMetaQualifier).stream()
+                    .filter(metaQualifier ->
+                            metaQualifier.getName().equals(Qualifier.REPO) && metaQualifier.getContent().isPresent())
+                    .forEach(repoQualifier ->
+                            reposToOpen.add(repoQualifier.getContent().get()));
+        });
+
+        reposToOpen.forEach(this::openRepositoryFromFilter);
+    }
+
+    /**
+     * Given a list of filter expressions, determine issues within the model that require a metadata update.
+     *
+     * @param filterTexts Filter expressions to process for metadata requests.
+     * @return Repo IDs and the corresponding issues in the repo requiring a metadata update.
+     */
+    private Map<String, List<TurboIssue>> tallyMetadataUpdate(List<FilterExpression> filterTexts) {
+        HashMap<String, List<TurboIssue>> toUpdate = new HashMap<>();
+
+        List<TurboIssue> allModelIssues = models.getIssues();
+
+        filterTexts.stream().filter(Qualifier::hasUpdatedQualifier).forEach(filterText -> {
+            List<TurboIssue> issuesInPanel = allModelIssues.stream()
+                    .filter(issue -> Qualifier.process(models, filterText, issue))
+                    .collect(Collectors.toList());
+
+            issuesInPanel.forEach(issue -> {
+                List<TurboIssue> issuesInRepo = toUpdate.get(issue.getRepoId());
+                if (issuesInRepo != null) {
+                    // If eixsts, just add the issue
+                    issuesInRepo.add(issue);
+                } else {
+                    // If not exists, create first then add
+                    issuesInRepo = new ArrayList<>();
+                    issuesInRepo.add(issue);
+                    toUpdate.put(issue.getRepoId(), issuesInRepo);
+                }
+            });
+        });
+
+        return toUpdate;
+    }
+
+    /**
+     * Retrieves metadata for given issues from the repository source, and then processes them for non-self
+     * update timings.
+     *
+     * @param repoId The repository containing issues to retrieve metadata for.
+     * @param issues Issues sharing the same repository requiring a metadata update.
+     * @return True if metadata retrieval was a success, false otherwise.
+     */
+    public CompletableFuture<Boolean> getIssueMetadata(String repoId, List<TurboIssue> issues) {
+        String message = "Getting metadata for " + repoId + "...";
+        logger.info("Getting metadata for issues "
+                + issues.stream().map(TurboIssue::getId).map(Object::toString).collect(Collectors.joining(", ")));
```

Comment:
Yup, probably is. The difference is that now there's no square bracket around the issues? lol